### PR TITLE
fixes issue #82 

### LINKED ISSUE #82: bug in makemore_part3_bn.ipynb
In this notebook the cell 20 is run without changing layer.training to `False` for all batchnorm layers. 

cell 20 is copied below:

```
# sample from the model
g = torch.Generator().manual_seed(2147483647 + 10)

for _ in range(20):
    
    out = []
    context = [0] * block_size # initialize with all ...
    while True:
      # forward pass the neural net
      emb = C[torch.tensor([context])] # (1,block_size,n_embd)
      x = emb.view(emb.shape[0], -1) # concatenate the vectors
      for layer in layers:
        x = layer(x)
      logits = x
      probs = F.softmax(logits, dim=1)
      # sample from the distribution
      ix = torch.multinomial(probs, num_samples=1, generator=g).item()
      # shift the context window and track the samples
      context = context[1:] + [ix]
      out.append(ix)
      # if we sample the special '.' token, break
      if ix == 0:
        break
    
    print(''.join(itos[i] for i in out)) # decode and print the generated word

```
a potential fix is to include the missing code fragment after the generator object is created.

### PR DESCRIPTION
**Pull request by kradkahaddi**
fixes #82 with batchnorm running average bug by setting layer.training=False in relevant cells (training loop, split_loss() function, and name generation code)

### CODE REVIEW

**Comment by kradkahaddi** [comment]
layer.training is set to True in the before the training loop to ensure no new bugs arise
